xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/dropdown-menu.html

```diff
@@ -1,26 +1,96 @@
 <template>
-  <div class="livedata-dropdown btn-group">
+  <div class="livedata-topbar">
```

Comment:
It's not a top bar as the one I first proposed, but I thought we still wanted to have a place where to have the dropdown menu, the search input, and the pagination, available by default across layouts. And that would be what's inside the dropdown menu that could be overridden if needed by a layout.
(Also I know there would be another dropdown menu associated with properties that would enable the user to add / remove columns, etc. when designing the layout, and this menu should be independent of anything)
If that's not what we finally stated, my bad, I'll update the design.

Reply:
> I thought we still wanted to have a place where to have the dropdown menu, the search input, and the pagination, available by default across layouts.

It's still a top bar. IMO each layout should be allowed to place the drop down menu, the pagination, the search, etc. wherever they want. A timeline layout might not need the pagination and it may want to integrate the drop down somewhere inside the layout. Instead of asking the layout to remove / hide what it doesn't need I'd rather allow them to place each component wherever they want.

> Also I know there would be another dropdown menu associated with properties that would enable the user to add / remove columns, etc. when designing the layout

No, the way I understood it is that we're going to have a **single** hamburger drop down menu, not two.

Reply:
Np, I can put the dropdown menu outside the bar, and create a bar (or something else) in each layout. However I'm not a big fan of letting each layout implement their user interactions buttons to different places, imo if a user wants to perform a basic action, he should have to go to same place to do it. Changing the position of basic actions might let more creativity for the layout design, but this is not user friendly, as the user will have to search each time the same action to different places. 

> No, the way I understood it is that we're going to have a single hamburger drop down menu, not two.

That's not what we decided at the end of our last call with Vincent. I thought we wanted to have two kinds of dropdowns, one general for common action of the Livedata (changing layout, filtering, sorting, exporting, ...), and one other for each property of the Livedata, that appears when we are able to modify its structure (and that comes with operations like: changing property type).

Reply:
> if a user wants to perform a basic action, he should have to go to same place to do it.

Yes, and that is the purpose of the drop down. This is the place the user will look for actions common to all layouts. There's no need for a toolbar, unless the layout has some special needs.

> and one other for each property of the Livedata, that appears when we are able to modify its structure

Yes, but that is not outside of the layout. That drop down is contextual, near the property, so inside the layout. For the card layout this will be on the card, for the table layout this will be inside the table header. That's what I understood.